lib/dataloader.ex

```diff
@@ -115,27 +158,33 @@ defmodule Dataloader do
   end
 
   @spec get(t, source_name, any, any) :: any | no_return()
-  def get(loader, source, batch_key, item_key) do
+  def get(loader = %Dataloader{options: options}, source, batch_key, item_key) do
     loader
     |> get_source(source)
     |> Source.fetch(batch_key, item_key)
-    |> do_get
+    |> do_get(options[:get_policy])
   end
 
-  defp do_get({:ok, val}), do: val
-  defp do_get(:error), do: nil
-
   @spec get_many(t, source_name, any, any) :: [any] | no_return()
-  def get_many(loader, source, batch_key, item_keys) when is_list(item_keys) do
+  def get_many(loader = %Dataloader{options: options}, source, batch_key, item_keys)
+      when is_list(item_keys) do
     source = get_source(loader, source)
 
     for key <- item_keys do
       source
       |> Source.fetch(batch_key, key)
-      |> do_get
+      |> do_get(options[:get_policy])
     end
   end
 
+  defp do_get({:ok, val}, :raise_on_error), do: val
+  defp do_get({:ok, val}, :return_nil_on_error), do: val
+  defp do_get({:ok, val}, :tuples), do: {:ok, val}
+
+  defp do_get({:error, reason}, :raise_on_error), do: raise(Dataloader.GetError, inspect(reason))
+  defp do_get({:error, _reason}, :return_nil_on_error), do: nil
+  defp do_get({:error, reason}, :tuples), do: {:error, reason}
+
```

Comment:
FYI - pretty much all changes in this PR revolve around allowing us to do this ^^

Reply:
I approve of this message.